### PR TITLE
Feature/sample

### PR DESCRIPTION
**Pull request by marto83**
Changes:
* Sample MQTT application
* Add comments to all public methods
* Updated README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,149 @@
...
+6. Create a new Pull Request
+
+## Release Process
+
+- Make sure the tests are passing in ci for the branch you're building
+- Update the CHANGELOG.md with any customer-affecting changes since the last release
+
+## License
+
+Copyright (c) 2019 Ably Real-time Ltd, Licensed under the Apache License, Version 2.0.  Refer to [LICENSE](LICENSE) for the license terms.
```
Comment (QuintinWillison) [inline README.md]: FWIW, 2020.

File: IO.Ably.DeltaCodec.SampleApp/Program.cs [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+namespace IO.Ably.DeltaCodec.SampleApp
+{
+    class Program
+    {
+        public static string ChannelName = "simple-app-mqtt" + Guid.NewGuid().ToString().Split("-").First();
+
+        static async Task Main(string[] args)
+        {
+            //HookupLogger();
```
[the remark this answers is not in the document]
Comment (marto83) [reply IO.Ably.DeltaCodec.SampleApp/Program.cs]: I left it on purpose so people can see how to setup Logging. Now thinking about it I should have put a comment next to it.

File: IO.Ably.DeltaCodec.SampleApp/Program.cs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+            bool isBinarySample = args.Length > 0 && args[0] == "binary";
+
+            var factory = new MqttFactory();
+            var consumer = new Consumer(factory);
+            var producer = new Producer(factory, isBinarySample);
+
+            await consumer.Start();
+            Task.Run(() => producer.Start());
+
+            Console.ReadLine();
```
[the remark this answers is not in the document]
Comment (marto83) [reply IO.Ably.DeltaCodec.SampleApp/Program.cs]: This is just to block the console app to finish executing. I'll add a message to it

File: IO.Ably.DeltaCodec.SampleApp/Program.cs [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        }
+
+        private static void HookupLogger()
```
[the remark this answers is not in the document]
Comment (marto83) [reply IO.Ably.DeltaCodec.SampleApp/Program.cs]: I want to leave it so people can easily enable logging for the transport. I've added comments.

File: README.md (hunk 1 above)
Comment (marto83) [reply README.md]: Done :)

File: IO.Ably.DeltaCodec.SampleApp/Sample.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using MQTTnet;
+using MQTTnet.Client;
+using MQTTnet.Client.Options;
+
+namespace IO.Ably.DeltaCodec.SampleApp
+{
+    public class ConsumerSample
```
[the remark this answers is not in the document]
Comment (marto83) [reply IO.Ably.DeltaCodec.SampleApp/Sample.cs]: I've added documentation. One file is what I used in the Readme. This way I definitely know it compiles

File: IO.Ably.DeltaCodec.SampleApp/Program.cs [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+        public async Task Start()
+        {
+            Console.WriteLine("Connecting Consumer");
+            await _client.ConnectAsync(_options);
+            await _client.SubscribeAsync(new TopicFilterBuilder().WithTopic(ChannelName).Build());
+            Console.WriteLine("### SUBSCRIBED ###");
+
+        }
+
+        private void OnSubscriberMessageReceived(MqttApplicationMessageReceivedEventArgs x)
```
[the remark this answers is not in the document]
Comment (marto83) [reply IO.Ably.DeltaCodec.SampleApp/Program.cs]: Done

File: IO.Ably.DeltaCodec.SampleApp/Program.cs [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+                117, 108, 97, 46, 19, 56, 1, 59, 0
+            },
+        };
+
+        public Producer(MqttFactory factory, bool isBinary)
+        {
+            _options = new MqttClientOptionsBuilder()
+                .WithClientId("producer")
+                .WithTcpServer("mqtt.ably.io", 8883)
+                .WithCredentials("oFpaLg.mB7oiw", "WP5kW-Mrk96MTaFq")
```
[the remark this answers is not in the document]
Comment (marto83) [reply IO.Ably.DeltaCodec.SampleApp/Program.cs]: Done

**Comment by marto83** [comment]
@QuintinWillison I'm going to merge this PR because it's blocking a ably-dotnet build (it's caused by the linter which complains about missing comments). I'm going to keep the branch open just in case you have further comments.

**Pull request merged**